daffodil-core/src/main/scala/org/apache/daffodil/core/dsom/ParticleMixin.scala

```diff
@@ -91,8 +91,15 @@ trait RequiredOptionalMixin { self: ElementBase =>
     val res = {
       if (isScalar) !this.isOutputValueCalc
       else if (isOptional) false
-      // Treat all arrays as non-required so that we can tolerate invalid
-      // infosets where the number of events is fewer than the array minimum occurrences.
+      // Treat all arrays as non-required if it has a default value or is optional
+      // so that we can tolerate invalid infosets where the number of events is
+      // fewer than the array minimum occurrences.
+      // But we don't want to unparse malformed data, so we only return true (required) if it is
+      // not defaultable (not yet implemented, so it will never be true at this point, but will
+      // produce a subset error with required arrays so people don't depend on broken/nyi behavior)
+      // and minOCcurs > 0
```

Comment:
This comment feels a bit confusing, we first say  we treat all arrays as optional, then say we don't actually want to do because we don't want to create malformed data. So the two sentences kindof conflict. What if we instead just say something like

> Arrays with minOccurs = 0 are not required to b in the infoset. Arrays that have at least one require element (i.e. minOccurs > 0) are required to have unparser events. Otherwise we could unparser malformed data. The exception to this is required arrays that are defaultable (which is not yet implemented, so will never be true at this point but will produce a subset error so people don't depend on broken/nyi behavior).